### PR TITLE
docs: foundational work

### PR DESCRIPTION
**Pull request by Nick-Lucas**
This PR will do a few things:

* Collapse all "other" sections by default to give a clearer overview of the content to first time readers
* Tweak section headings to mean more to new users (ie. @trpc/server -> Backend Usage)
* Reorganise some content which does not belong, into its proper place (ie. aborting-procedures broken up into client and react pages, ie. move react/next usage links into their sections) - this is not exhaustive, most of the docs are currently written mixing things up, it's going to take multiple PRs to unwind in a nice way
* Add Introduction pages to all sections
* Revise and rename Setup guides for all sections

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?

## ✅ Checklist

- [ ] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: www/docs/client/aborting-procedures.md [hunk 1]
```diff
@@ -2,49 +2,9 @@
 id: aborting-procedure-calls
 title: Aborting Procedure Calls
 sidebar_label: Aborting Procedure Calls
-slug: /aborting-procedure-calls
+slug: /client/aborting-procedure-calls
```
Comment (KATT) [inline www/docs/client/aborting-procedures.md:5]: Seems a bit arbitrary to change this? I don't mind it, but then we should go through all "client"-stuff and rename them. Also we should add a redirect

tl;dr - I think the url change shouldn't be part of this PR and instead done on later PR that changes _all_ routes

File: www/docs/client/introduction.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# tRPC Client
```
Comment (KATT) [inline www/docs/client/introduction.md:1]: Aren't we missing the headers etc here? Maybe it's not needed

File: www/docs/reactjs/introduction.mdx [hunk 3]
```diff
@@ -1,182 +1,41 @@
 ---
 id: introduction
-title: Usage with React
-sidebar_label: Usage with React
-description: How to use and setup tRPC in React
-slug: /react
+title: React Query Integration
+sidebar_label: React Query Integration
+description: React Query Integration
+slug: /reactjs/introduction
```
Comment (KATT) [inline www/docs/reactjs/introduction.mdx:6]: See previous comment on slugs

File: www/docs/reactjs/setup.mdx [hunk 4]
```diff
@@ -0,0 +1,128 @@
+---
+id: setup
+title: Usage with React
+sidebar_label: Usage with React
+description: How to use and setup tRPC in React
+slug: /reactjs/setup
```
Comment (KATT) [inline www/docs/reactjs/setup.mdx:6]: see prev comment

File: www/vercel.json [hunk 5]
```diff
@@ -1,5 +1,20 @@
...
+    {
+      "source": "/docs/aborting-procedure-calls",
+      "destination": "/docs/client/aborting-procedure-calls",
+      "permanent": true
+    },
+    {
+      "source": "/docs/client/vanilla",
+      "destination": "/docs/client/setup",
+      "permanent": true
+    },
```
Comment (KATT) [inline www/vercel.json:17]: noice

**Review by KATT — COMMENTED** [review]
Didn't read the actual text (I trust you and the gang here!), just did some sanity checking and I think URL changes should be done as it's own PR where we blitz through _everything_

**Comment by Nick-Lucas** [comment]
Can definitely pull them up into an initial merge once we're happy, the reason for some changes is that we have duplicate URLs after splitting overloaded pages into integration specific ones with the same name, but some changes because of the intro/setup pattern I'm introducing. 

aborting-procedures is now 2 pages for an example of the former case

File: www/docs/client/introduction.md (hunk 2 above)
Comment (Nick-Lucas) [reply www/docs/client/introduction.md:1]: Was thinking the same, react docs have it. Will do

File: www/docs/client/aborting-procedures.md (hunk 1 above)
Comment (Nick-Lucas) [reply www/docs/client/aborting-procedures.md:5]: Just to pop deets here, yes let's separate out the router changes for existing pages into an initial PR, I will do it once I'm happy with this initial re-org and know the full scope :)

For the "why" - we have URL conflicts now, because categorisation and consistency are important. This page doesn't only document the vanilla client so I've moved the react stuff to the react category

File: www/docs/nextjs/setup.mdx [hunk 6]
```diff
@@ -209,12 +204,14 @@ export const trpc = createTRPCNext<AppRouter>({
 });
 ```
 
-:::note
-`createTRPCNext` does not work with interop mode. If you are migrating from v9 using interop, keep using [the old way of initializing tRPC](../../versioned_docs/version-9.x/nextjs/introduction.md#4-create-trpc-hooks).
+:::warning
```
Comment (KATT) [inline www/docs/nextjs/setup.mdx:207]:
```suggestion
:::note
```

We'll kill v9 soon, no need to warn

File: www/docs/reactjs/setup.mdx [hunk 7]
```diff
@@ -1,84 +1,17 @@
 ---
 id: setup
-title: Usage with React
-sidebar_label: Usage with React
+title: Set up the React Query Integration
```
Comment (KATT) [inline www/docs/reactjs/setup.mdx:3]:
```suggestion
title: Setup the React Query Integration
```

File: www/docs/client/setup.mdx [hunk 8]
```diff
@@ -0,0 +1,83 @@
+---
+id: setup
+title: Set up a tRPC Client
```
Comment (KATT) [inline www/docs/client/setup.mdx:3]:
```suggestion
title: Setup a tRPC Client
```

Comment (Nick-Lucas) [reply www/docs/client/setup.mdx:3]: What do you think of this? 

https://writingexplained.org/setup-vs-set-up-difference#:~:text=Both%20words%20have%20different%20meanings,in%20order%20or%20arranging%20them.

Comment (KATT) [reply www/docs/client/setup.mdx:3]: If it's conscious please ignore! :D

Comment (Nick-Lucas) [reply www/docs/client/setup.mdx:3]: Ha no worries, yes I believe this is grammatically correct, but English is subtle so I'm happy to be convinced otherwise 🙂

File: www/docs/nextjs/setup.mdx [hunk 9]
```diff
@@ -96,9 +87,10 @@ If strict mode is too harsh, you'll at least want to enable `strictNullChecks`:
 
 ### 3. Create a tRPC router
 
-Initialize your tRPC backend using the `initTRPC` function and create your first router. We're going to make a simple "hello world" router and procedure here - but be sure to browse the rest of the docs for information about creating more complex APIs.
+Initialize your tRPC backend in `src/server/trpc.ts` using the `initTRPC` function, and create your first router. We're going to make a simple "hello world" router and procedure here - but for deeper information on creating your tRPC API you should refer back to the [quickstart guide](/docs/quickstart) and the [Backend usage docs](/docs/server/procedures)
```
Comment (juliusmarminge) [inline www/docs/nextjs/setup.mdx:90]: just chose an arbitrary link for the backend usage as it was broken and we dont have an introduction/setup page for that. should we add one of those?

File: www/docs/nextjs/introduction.mdx [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+sidebar_label: Next.js Integration
+slug: /nextjs/introduction
+---
+
+### tRPC ❤️ Next.js
+
+Next.js makes it easy to build a client and server together in one codebase. tRPC makes it easy to share types between them, ensuring typesafety for your application's data fetching.
+
+Our Next.js integration combines our [React Query Integration](/docs/reactjs/introduction) with some Next.js specific APIs, to handle client-side and SSR needs.
+
```
[the remark this answers is not in the document]
Comment (Nick-Lucas) [reply www/docs/nextjs/introduction.mdx:13]: Is this something you'd like to take a look at? My Next.js knowledge is very weak so I've tried not to do too much here, but it needs a lot more like you suggest

File: www/docs/reactjs/introduction.mdx [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+---
+
+tRPC offers a first class integration with React. Under the hood this is simply a wrapper around the very popular [@tanstack/react-query](https://tanstack.com/query/latest), so we recommend that you familiarise yourself with React Query, as their docs go in to much greater depth on its usage.
+
+:::tip
+If you are using Next.js we recommend using [our integration with that](/docs/nextjs/introduction) instead
+:::
+
```
[the remark this answers is not in the document]
Comment (Nick-Lucas) [reply www/docs/reactjs/introduction.mdx:14]: Yep, intro should give an idea of the solved problems and usage, setup gives the depth :)

File: www/docs/reactjs/aborting-procedures.md [hunk 12]
```diff
@@ -0,0 +1,40 @@
+---
+id: aborting-procedure-calls
+title: Aborting Procedure Calls
+sidebar_label: Aborting Procedure Calls
+slug: /reactjs/aborting-procedure-calls
+---
+
+By default, tRPC does not cancel requests on unmount. If you want to opt into this behaviour, you can provide `abortOnUnmount` in your configuration.
```
Comment (juliusmarminge) [inline www/docs/reactjs/aborting-procedures.md:8]: Does react query have a docs link on what this means? I think TKDodo said something about this a while back

File: www/docs/nextjs/setup.mdx (hunk 9 above)
Comment (Nick-Lucas) [reply www/docs/nextjs/setup.mdx:90]: I think we can link to `/server/introduction` and then it's a sensible starting point with the generated index, but also sensible if we decide to write something up properly there

File: www/docs/reactjs/aborting-procedures.md (hunk 12 above)
Comment (Nick-Lucas) [reply www/docs/reactjs/aborting-procedures.md:8]: Honestly I didn't rewrite this, it's just split out from the other file. Though we need to do a full pass on breaking up the `server/` pages into the integrations as they're very overloaded right now and it's a key cause of users struggling to find the content they want (it's technically in the wrong place!)

File: www/docs/client/setup.mdx [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+Under the hood this creates a typed [JavaScript Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) which allows you to interact with your tRPC API in a fully type-safe way:
+
+```ts title='client.ts'
+const bilbo = await client.getUser.query('id_bilbo');
+// => { id: 'id_bilbo', name: 'Bilbo' };
+
+const frodo = await client.createUser.mutate({ name: 'Frodo' });
+// => { id: 'id_frodo', name: 'Frodo' };
+```
+
```
[the remark this answers is not in the document]
Comment (Nick-Lucas) [reply www/docs/client/setup.mdx:82]: Not sure, I lean toward "no" just because "how does this work internally?" isn't likely to be a top question at this stage of the docs flow - the question will be "how do I set this up?"

Now, is there a place for it? I think yes, so I'll add a note into _the plan_

File: www/docs/reactjs/aborting-procedures.md (hunk 12 above)
Comment (Nick-Lucas) [reply www/docs/reactjs/aborting-procedures.md:8]: Looks like it is a bespoke property by the way, it's not a RQ thing, so makes sense to document it here

File: www/docs/nextjs/introduction.mdx (hunk 10 above)
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply www/docs/nextjs/introduction.mdx:19]: added some words but would be nice with a second pair of eyes. cc @KATT

File: www/docs/reactjs/aborting-procedures.md [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+import type { AppRouter } from '../server/router';
+
+export const trpc = createTRPCReact<AppRouter>();
+
+function AppProvider() {
+  const [client] = useState(() =>
+    trpc.createClient({
+      links: [],
+      // FIXME: fix this in core
+      // abortOnUnmount: true, // 👈
```
Comment (juliusmarminge) [inline www/docs/reactjs/aborting-procedures.md:43]: remove fixme and uncomment once fixed in core

File: www/docs/client/setup.mdx (hunk 13 above)
[the remark this answers is not in the document]
Comment (KATT) [reply www/docs/client/setup.mdx:82]: Agreed with @Nick-Lucas

File: www/docs/nextjs/introduction.mdx (hunk 10 above)
[the remark this answers is not in the document]
Comment (KATT) [reply www/docs/nextjs/introduction.mdx:19]: It's great as-is, we can always iterate

**Review by KATT — APPROVED** [review]
